src/autopas/containers/hierarchicalGrids/HierarchicalGrids.h

```diff
@@ -0,0 +1,644 @@
+
+    return _carrierLevel.deleteParticle(cellIndex, particleIndex);
+
+  }
+
+  /**
+   * @brief Get the Hierarchy Level Of Particle object
+   * 
+   * @param p 
+   * @return Level the particle should be sorted into. 
+   */
+  unsigned int getHierarchyLevelOfParticle(const ParticleType &p) {
+
+    // If only one HG level is available all particles are sorted into this level.
+    if (_numberOfLevels == 0) {
+      return 0;
+    }
+        
+    // Get the radius of the particle in question
+    auto particleRad = p.getRad();
+
+    unsigned int level;
+
+    // Loop over all hierarchy levels
+    for (level = 0; level < _numberOfLevels; level++) {
+      // As long as the '_hierarchicalGridBorders' element is smaller than the radius continue to cycle
+      if (particleRad > _hierarchicalGridBorders[level]) break;     
+    }
+
+    if(level == 0) {
+    utils::ExceptionHandler::exception(
+      "HierarchicalGrids: Trying to get the hierarchy level of a particle which is too large to fit in the grid.\n"
+      "Particle radius {}\n"
+      "Upper limit of the largest level {}",
+      particleRad, _hierarchicalGridBorders[level]);
+    }
+    // Break and return the level
+    return level - 1;
+  }
+
+  void setHierarchicalGridBorders(std::vector<double> boundaries) {
+
+    _hierarchicalGridBorders = boundaries;
+
+  }
+
+  std::vector<double> getHierarchicalGridBorders() {
+
+    return _hierarchicalGridBorders;
+
+  }
+
+
+  private:
+
+  /**
+   * @brief Get the number Of cross-levels of the H-Grid
+   * 
+   * @param levels 
+   * @return unsigned int 
+   */
+  unsigned int getNumberOfCrossLevels(unsigned int levels) {
+
+    unsigned int numberOfCrossLevels = 0;
+    for (unsigned int i = 1; i < levels; i++)
+    {
+      numberOfCrossLevels += i;
+    }
+    
+    return numberOfCrossLevels;
+  }
+
+  /**
+  * @brief Vector containing the HGrid's different hierarchy levels of Linked Cells.
+  * 
+  */
+  std::deque<LinkedCells<Particle>> _hierarchyLevels;
+
+  /**
+   * @brief Carrier level containing all particles for functions like addParticle() etc.
+   * 
+   */
+  autopas::LinkedCells<Particle> _carrierLevel;
+
+  /**
+   * @brief Number of levels (hierarchies) in the H-Grid
+   * 
+   */
+  unsigned int _numberOfLevels;
+
+  std::array<double, 2> _particleRadiusRange;
+
+  std::array<double, 3> _boxMin; 
+  std::array<double, 3> _boxMax; 
+  double _cutoff;
+  double _skinPerTimestep; 
+  unsigned int _rebuildFrequency;
+  double _cellSizeFactor;
+  autopas::LoadEstimatorOption _loadEstimator;
+
+  /**
+   * @brief Vector containing the upper boundaries of the grid levels. 
+   * 
+   * For example, the first element describes the upper boundary of the largest H-Grid level,
+   * i.e. the largest particles fitting in the topmost level.
+   * By design '_hierarchicalGridBorders' should have the same '_numberOfLevels' as the H-Grid.
+   * 
+   */
+  std::vector<double> _hierarchicalGridBorders;
```

Comment:
Why is this needed? Is `_particleRadiusRange` not enough?

Reply:
Now after reading most of the class it seems `_particleRadiusRange` is not used at all? If that is correct, get rid of it or get rid of `_hierarchicalGridBorders` but sooner or later, I think the mechanism that decides which particle needs to go where might need to be reconsidered (see the comment in the constructor about the radii variables).

Reply:
Got rid of _particleRadiusRange. Idea behind it was to account for particle radii (but I did not do it in the end).
I fully agree that there needs to be a more sophisticated logic behind the sorting and level size mechanisms.